### ISSUE TITLE
[Epic] Finish Reclocking

### ISSUE THREAD

**Issue by nmeagan11** [MaterializeInc/materialize#12867]
### Initiative and Theme

Materialize is Dependable; Materialize makes observability easy

### Problem

[Reclocking](https://github.com/MaterializeInc/materialize/issues/11218) was mostly finished in M1, but we have outstanding eng tasks. Most notably is https://github.com/MaterializeInc/materialize/issues/11908, which is necessary for progress monitoring. Additionally, though reclocking data is now stored in persist, STORAGE doesn't expose it to ADAPTER.

### Success Criteria

The use cases from the [product spec](https://github.com/MaterializeInc/materialize/issues/11218#issuecomment-1087849617) (except BYO) are solved.

### Time Horizon

Small

### Blockers

None

### Blocks

Low priority aspects of https://github.com/MaterializeInc/cloud/issues/3259 related to source/view lag
Real Time Recency: https://github.com/MaterializeInc/materialize/issues/11531

```[tasklist]
### Tasks

- [ ] https://github.com/MaterializeInc/materialize/issues/11908
- [x] https://github.com/MaterializeInc/materialize/issues/12267
- [ ] https://github.com/MaterializeInc/materialize/pull/15813
- [ ] #17788
```

**Comment by uce** [comment]
Relevant discussion to follow: https://github.com/MaterializeInc/materialize/issues/14266#issuecomment-1247293118

**Comment by sploiselle** [comment]
This work got sidetracked in implementing some necessary hardening for dropping resources in the storage controller (#17105); this got delayed because I discovered some unexpected missing features in the storage controller, but reduced the scope of this to be only as much as is required for supporting queryable remap shards.

Once that lands, I have another PR that actually introduces queryable remap collections (#15813).

**Comment by sploiselle** [comment]
#15813 has an occasionally green set of commits that implements this feature. Given the size of the PR, I am trying to incrementally add the commits to make reviews easier and debugging failures that creep in simpler to debug. The list of atomized PRs so far is:
- [] #17105
- [] #17284
- [] #17287

**Comment by sploiselle** [comment]
#15813 is up for review with this feature.

**Comment by sploiselle** [comment]
There are some questions around the interface we ought to use to express a dependency between the sinces of collections; I have a draft PR up with work from Aljoscha #17590. If this approach suffices it has the possibility of closing out the majority of the pending feedback on https://github.com/MaterializeInc/materialize/pull/15813

**Comment by sploiselle** [comment]
Implementing this is now waiting on some feedback for how to handle adding progress collections to existing sources https://github.com/MaterializeInc/materialize/pull/15813#issuecomment-1431912480

**Comment by sploiselle** [comment]
#15813 is green and awaiting a final review

**Comment by sploiselle** [comment]
https://github.com/MaterializeInc/materialize/pull/15813 has merged; the only remaining issue is to document the feature.

**Comment by sploiselle** [comment]
This is done as of #17788
